src/libraries/System.Text.Json/tests/Serialization/Object.ReadTests.cs

```diff
@@ -278,15 +278,15 @@ private class CollectionWithoutPublicParameterlessCtor : IList
 
             internal CollectionWithoutPublicParameterlessCtor()
             {
-                Debug.Fail("The JsonSerializer should not be callin non-public ctors, by default.");
+                Debug.Fail("The JsonSerializer should not be calling non-public ctors, by default.");
             }
 
             public CollectionWithoutPublicParameterlessCtor(List<object> list)
             {
                 _list = list;
             }
 
-            public object this[int index] { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+            public object this[int index] { get => _list[index]; set => _list[index] = value; }
```

Comment:
Is the indexer called now? Not sure why this was changed.

Reply:
> Is the indexer called now?

Yes, per the change in this method: https://github.com/dotnet/runtime/pull/39363/files#diff-f238cfd315fa616a94be1d5f58c786e0R52. I made this change to mitigate perf regressions with serializing `IList` types.